Integration Vision update (#1)
HarshaNalluru/amqp-log-explainer: PROJECT_PROPOSAL.md

## Project Goals

> - Provide a backend library (in genAIScript) for AMQP log analysis, designed for integration in multiple environments (CLI, UI(react app), CI, MCP, etc.).
> - Enable users and systems to analyze large AMQP log files efficiently.
> - Support AI-generated summaries and detailed insights for each logical chunk of the log.
> - Offer both CLI tooling and an optional interactive UI for exploring log data.

Author: ```suggestion
- Provide a backend library (in genAIScript) for AMQP log analysis, designed for integration in multiple environments (CLI, UI(react app), CI, MCP, etc.).
- Enable users and systems to analyze large AMQP log files efficiently.
- Support AI-generated summaries and detailed insights for each logical chunk of the log.
- Investigate higher-level APIs to process the logs efficiently, avoiding manual chunking.
- Offer both CLI tooling and an optional interactive UI for exploring log data.
```

> - Support natural language queries about the logs.

Author: ```suggestion
- Support natural language queries about the logs and insights.
```

> - Offer both CLI tooling and an optional interactive UI for exploring log data.

Author: ```suggestion
- Offer both CLI tooling and an interactive UI for exploring logs.
```

> - Make the architecture extensible for integration with Azure SDK MCP server, fault-injector, and other tools.
> - Investigate higher-level APIs to process the logs efficiently, avoiding manual chunking.

Author: ```suggestion
- Make the architecture extensible for integration with Azure SDK MCP server, fault-injector, and other tools.
```

- Offer both CLI tooling and an optional interactive UI for exploring log data.
- Support natural language queries about the logs.
- Make the architecture extensible for integration with Azure SDK MCP server, fault-injector, and other tools.
- Investigate higher-level APIs to process the logs efficiently, avoiding manual chunking.

## Use Cases

- **Developer CLI:** Analyze logs locally or in CI pipelines, outputting structured summaries and error reports.
- **Automated CI Analysis:** Automatically generate and dump log analysis for stress test runs, reducing the time taken in a manual review.
- **Azure SDK(internal) MCP Integration:** Provide in-context log insights and troubleshooting within the MCP server/VS Code.
- **UI(react app) for Exploration:** Rich, interactive UI for deep-dive analysis and demo purposes.

## Current Capabilities

- React-based UI for interactive exploration. Upload and view raw AMQP log files in the browser.
- Log file is split into manageable chunks and sent to Azure OpenAI for analysis.
- Each chunk is summarized with its purpose and key events/errors in a structured format.
- Expandable UI for exploring chunk and subchunk insights.

> - Query support for asking questions about the logs in natural language.

Author: ```suggestion
- Query support for asking questions about the logs in natural language.
- Dynamic, real-time rendering as logs are analyzed.
```

## Project Goals

> - Provide a reusable backend library (in genAIScript) for AMQP log analysis, designed for integration in multiple environments (CLI, UI(react app), CI, MCP, etc.).

Author: ```suggestion
- Provide a backend library (in genAIScript) for AMQP log analysis, designed for integration in multiple environments (CLI, UI(react app), CI, MCP, etc.).
```

- Investigate higher-level APIs to process the logs efficiently, avoiding manual chunking.

## Use Cases

- **Developer CLI:** Analyze logs locally or in CI pipelines, outputting structured summaries and error reports.
- **Automated CI Analysis:** Automatically generate and dump log analysis for stress test runs, reducing the time taken in a manual review.
- **Azure SDK(internal) MCP Integration:** Provide in-context log insights and troubleshooting within the MCP server/VS Code.
- **UI(react app) for Exploration:** Rich, interactive UI for deep-dive analysis and demo purposes.

## Current Capabilities

- React-based UI for interactive exploration. Upload and view raw AMQP log files in the browser.
- Log file is split into manageable chunks and sent to Azure OpenAI for analysis.
- Each chunk is summarized with its purpose and key events/errors in a structured format.
- Expandable UI for exploring chunk and subchunk insights.
- Query support for asking questions about the logs in natural language.
- Dynamic, real-time rendering as logs are analyzed.
- Support for multiple log formats and comparison of logs from different SDKs/languages.

## Workflow

> - The current prototype is built with React and TypeScript, but the core logic will be migrated to a reusable backend library.

Author: ```suggestion
- The current prototype is built with React and TypeScript, but the core logic will be migrated to a backend library.
```

- Uses Azure OpenAI via `@azure-rest/ai-inference` for chunk analysis.

> - Log chunking is optimized for API limits (e.g., 32K/1M token models).

Author: ```suggestion
- Log chunking is optimized for API limits (e.g., 32K/1M token models).
- Insights are rendered in a tree structure for easy navigation.
```

- Offer both CLI tooling and an optional interactive UI for exploring log data.
- Support natural language queries about the logs.
- Make the architecture extensible for integration with Azure SDK MCP server, fault-injector, and other tools.
- Investigate higher-level APIs to process the logs efficiently, avoiding manual chunking.

## Use Cases

- **Developer CLI:** Analyze logs locally or in CI pipelines, outputting structured summaries and error reports.
- **Automated CI Analysis:** Automatically generate and dump log analysis for stress test runs, reducing the time taken in a manual review.
- **Azure SDK(internal) MCP Integration:** Provide in-context log insights and troubleshooting within the MCP server/VS Code.
- **UI(react app) for Exploration:** Rich, interactive UI for deep-dive analysis and demo purposes.

## Current Capabilities

- React-based UI for interactive exploration. Upload and view raw AMQP log files in the browser.
- Log file is split into manageable chunks and sent to Azure OpenAI for analysis.
- Each chunk is summarized with its purpose and key events/errors in a structured format.
- Expandable UI for exploring chunk and subchunk insights.

> - Query support for asking questions about the logs in natural language.

Author: ```suggestion
- Query panel for asking questions about the logs in natural language.
```

## Future Enhancements / Next Steps

1. **Investigate Higher-Level APIs:** Explore and adopt higher-level APIs to further simplify log analysis and reduce the need for manual chunking.
2. **GenAI Script Integration & Automation:** Integrate with GenAI scripts to automate and customize log analysis workflows, including automatic generation of analysis for our stress test logs in CI.
3. **Smarter Chunking & Context Preservation:** Implement log markers (e.g., "azure:" suffix), overlapping windows, and advanced chunking strategies to improve context and reduce API calls.
4. **Cost Optimization:** Batch requests and leverage higher token models (e.g., GPT-4.1 1M tokens) for fewer, larger requests, minimizing API costs.
5. **Advanced Preprocessing & Correlation:** Develop preprocessing to split and tag logs more intelligently, and algorithms for correlating non-consecutive events and errors across large AMQP logs.
6. **Enhanced Cross-Language Support:** Improve comparison and correlation of logs from different SDKs/languages for unified analysis.

> 7. **Azure SDK MCP Integration:** Connect with Maor's MCP server for deeper Azure insights and automated troubleshooting.

Author: ```suggestion
7. **Azure SDK MCP Integration:** Connect with Maor's MCP server for deeper Azure insights and automated troubleshooting without leaving VS Code.
```
